client/state/directly/actions.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Internal dependencies
+ */
+import {
+	initialize as initializeDirectly,
+	askQuestion as askDirectlyQuestion,
+	maximize as maximizeDirectly,
+	minimize as minimizeDirectly,
+	openAskForm as openDirectlyAskForm,
+} from 'lib/directly';
+import {
+	DIRECTLY_ASK_QUESTION,
+	DIRECTLY_INITIALIZE,
+	DIRECTLY_MAXIMIZE,
+	DIRECTLY_MINIMIZE,
+	DIRECTLY_OPEN_ASK_FORM,
+} from 'state/action-types';
+
+export const askQuestion = ( questionText, name, email ) => ( dispatch ) => {
+	dispatch( { type: DIRECTLY_ASK_QUESTION, questionText, name, email } );
+	askDirectlyQuestion( questionText, name, email );
+};
```

Comment:
> Ok — so is there a rule of thumb for when to use middleware instead of a thunk from an action creator for side-effecting operations?

why yes there is! **always use middleware** and **never use action thunks** :smile:

not kidding. if it's confusing please ask for help - we'll happily walk you through the process!

> I've seen some hairy situations where actions get dispatched from middleware and cause problems

sorry to hear that @omarjackman - this probably reveals issues with the logic more than the use of middleware. I'd be happy to help you sort those out if you want to revisit the code. thunks introduce unnecessary complexity and opaqueness to the otherwise elegant Redux pattern and they are a tempting but nefarious work-around to the system.